shtdlib.sh

```diff
@@ -2526,26 +2533,47 @@ function associate_array {
     done
 }
 
-# Safely loads config file
+# Safely loads config file or config from stdin
 # First parameter is filename, all consequent parameters are assumed to be
 # valid configuration parameters
+# By default settings are printed when config is piped since variables set
+# would only live in a subshell, this can be changed using the "print_settings"
+# variable
 function load_config {
-    config_file="${1}"
-    # Verify config file permissions are correct and warn if they aren't
-    # Dual stat commands to work with both linux and bsd
-    while read -r line; do
+    # If first argument is a filename read it and pass it to the function
+    if [ -f "${1:-}" ]; then
+        print_settings='false'
+        load_config "${@:2}" < "${1}"
+        return
+    elif [ -t 0 ] ; then
+        color_echo red "No config filename provided or data on stdin, exiting"
+        return 1
+    else
+        print_settings="${print_settings:-true}"
```

Comment:
I can't figure out when we ever reach this case...

Reply:
For some reason my reply is missing, think about the function being called iteratively ...